File: src/model/State.java
package model;

import java.awt.*;

public class State {

    public int getDimension() {
        return Dimension;
    }

    private int Dimension;
    private int [] board;
    private int cost;


    public State (int[] board){
        this.board = board;
        Dimension = board.length;
        this.cost=computeCost();
    }

    public int[] getBoard(){
        return this.board;
    }

    public void showBoard(){
        System.out.println("Board");
        System.out.println("-----");
        for (int i = 0; i < Dimension; i++) {
                System.out.printf(String.valueOf(board[i]));

        }
        System.out.printf(" Cost = "+cost);
        System.out.println();
    }

    public int getCost(){
        return cost;
    }
    private int computeCost(){
        int cost=0;
        for(int i=0;i<Dimension;i++){
            for (int j=i+1;j<Dimension;j++){
                if(isThreaten(new Point(i,board[i]),new Point(j,board[j]))){
                    cost+=1;
                }
            }
        }
        return cost;
    }

    private boolean isThreaten(Point a, Point b) {
        if(a.getX()==b.getX()){
            return true;
        }
        if(a.getY()==b.getY()){
            return true;
        }
        if(Math.abs(a.getX()-b.getX())==Math.abs(a.getY()-b.getY())){
            return true;
        }
        return false;
    }

}

File: N-Queens/src/model/State.py
class State:
    def __init__(self, initial):
        self.board = initial
        self.cost = self.computeCost()

    def computeCost(self):
        sum = 0
        for i in range(0, len(self.board)):
            for j in range(i + 1, len(self.board)):
                check = self.isThreaten(i, self.board[i], j, self.board[j])
                if check:
                    sum += 1
        return sum

    def isThreaten(self, i, param, j, param1):
        if i == j:
            return True
        if param == param1:
            return True
        if abs(i - j) == abs(param1 - param):
            return True
        return False
